src/flb_network.c

```diff
@@ -987,6 +985,7 @@ static struct flb_dns_lookup_context *flb_net_dns_lookup_context_create(
     lookup_context->finished = 0;
     lookup_context->dropped = 0;
     lookup_context->dns_ctx = dns_ctx;
+    MK_EVENT_ZERO(&lookup_context->response_event);
```

Comment:
This one shouldn't be here, it has to be initialized in [line 859](https://github.com/fluent/fluent-bit/blob/b4b3254dd77c58a25efa532bca272b27375bed2c/src/flb_network.c#L859) after the socket is created and in this case  `MK_EVENT_NEW` should be used for the moment (because we don't allow the DNS client to issue more than one connection which is one of the many design flaws)